providers/implementations/keymgmt/ec_kmgmt.c

```diff
@@ -341,12 +341,8 @@ int common_import(void *keydata, int selection, const OSSL_PARAM params[],
      */
     if ((selection & OSSL_KEYMGMT_SELECT_DOMAIN_PARAMETERS) == 0)
         return 0;
-    if ((selection & OSSL_KEYMGMT_SELECT_PRIVATE_KEY) != 0
-            && (selection & OSSL_KEYMGMT_SELECT_PUBLIC_KEY) == 0)
-        return 0;
 
-    if ((selection & OSSL_KEYMGMT_SELECT_DOMAIN_PARAMETERS) != 0)
-        ok = ok && ec_group_fromdata(ec, params);
+    ok = ok && ec_group_fromdata(ec, params);
```

Comment:
Did you notice the `include_private` further down, passed to `ec_key_fromdata()`?  That has lost its importance (I'm not sure why it was even important to begin with), so I'd throw that away, including removing that last `ec_key_fromdata()` argument.

Reply:
Doesn't that mean that we would no longer pay any attention to the `OSSL_KEYMGMT_SELECT_PRIVATE_KEY` selection parameter? i.e. we would only drive things off of whether the private key data was actually present in the params - regardless of what `selection` says. That doesn't seem quite right to me....and if we no longer consider that selection parameter important, then perhaps a bigger refactor is required?

Reply:
Saying that `selection` is unimportant if we don't specifically check for `OSSL_KEYMGMT_SELECT_PRIVATE_KEY` is a bit of an exaggeration, isn't it?  We do still check that a keypair is expected before calling `ec_key_fromdata()`, do we not?  However, I don't see the value with being that level of picky here if `ec_key_fromdata()` can discover on its own if there's a private key present in the data or not.

Reply:
Frankly, I'm quite reluctant to make the suggested change. I seriously dislike the inconsistency of paying attention to some of the section flags, but not others. To me that has a nasty smell about it. It also seems quite tangential to the point of this PR.

Reply:
[shrug]